providers/github/GitHubProvider.js

```
@@ -1,57 +1,428 @@
+import parse from 'parse-diff';
+
 import { HttpProvider } from '../http/HttpProvider.js';
+import { doGeminiResponse } from '../gemini/Gemini.js';
+import { Logger } from '../../lib/logger.js';
 
 export class GitHubProvider {
-    constructor() {
+    _AGENT_USER = process.env.GITHUB_AGENT_USER;
+
+    action;
+    raw;
+    repoUrl;
+    pullRequestUrl;
+    latestCommitHash;
+
+    description;
+    comments = [];
+    diff;
+
+    constructor(req, action) {
         this.http = new HttpProvider({
             Authorization: `Bearer ${process.env.GITHUB_TOKEN}`,
+            Accept: "application/vnd.github.v3+json"
         });
+
+        this.raw = req;
+        this.action = action;
+        this.repoUrl = req.body.repository.url;
+        this.pullRequestUrl = req.body.pull_request.url;
+        this.pullRequestId = req.body.pull_request.id;
+        this.latestCommitHash = req.body.after;
+
+        this.processRequest(req).catch((e) => Logger.error(e.message));
```

Running an async process from the constructor without `await` can hide errors from the main request handler. Consider an `init()` method instead. You can discuss the request or make a change to trigger a re-review. When all threads are resolved -- the agent will approve.

---

This `.catch` still results in a 'fire-and-forget' call. The main request handler won't know this async process failed. An `init()` method is safer.

---

This pattern remains. Kicking off an unawaited async process from the constructor can hide errors from the main request handler.

---

Great! Moving the async logic to `processRequest` and awaiting it in `index.js` is exactly what was needed. This resolves the issue. Thanks!